prov/sm2/src/sm2_coordination.c

```diff
@@ -312,6 +312,16 @@ ssize_t sm2_file_open_or_create(struct sm2_mmap *map_shared)
 		return -FI_EAVAIL;
 	}
 
+	/* Remap the file to be big enough to hold every region upfront.
+	 * This is required to get rid of race conditions that may have
+	 * previously occured when we tried to remap in sm2_av_insert(),
+	 * sm2_fifo_send() or sm2_fifo_recv().
+	 */
+	header = (struct sm2_coord_file_header *) map_shared->base;
+	max_file_size = header->ep_regions_offset +
+			header->ep_region_size * SM2_MAX_UNIVERSE_SIZE;
```

Comment:
Can we also reduce `SM2_MAX_UNIVERSE_SIZE`? With 2048, we would need about 8GB of memory for every run (2048 * 1024 * 4096) which seems a lot to me.

Reply:
I agree with @sunkuamzn. Now that we allocate the whole, we've got to play it safer with how much memory we anticipate.

Reply:
done